### PR TITLE
[flutter_tools] Migrate more tool tests to null-safety

### PR DESCRIPTION
**Pull request by christopherfujino**
Part of https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

**Comment by christopherfujino** [comment]
cc @eliasyishak

File: packages/flutter_tools/lib/src/commands/update_packages.dart [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ String generateFakePubspec(
 /// It ends up holding the full graph of dependencies, and the version number for
 /// each one.
 class PubDependencyTree {
-  final Map<String, String?> _versions = <String, String>{};
+  final Map<String, String?> _versions = <String, String?>{};
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/update_packages.dart:1442]: This was a bug, although apparently didn't get hit in regular usage

File: packages/flutter_tools/test/commands.shard/hermetic/config_test.dart [hunk 2]
```diff
@@ -288,7 +286,7 @@ class FakeAndroidSdk extends Fake implements AndroidSdk {
 
 class FakeFlutterVersion extends Fake implements FlutterVersion {
   @override
-  String channel;
+  late String channel;
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/config_test.dart:289]: I tried making this a getter backed by a nullable private, but the analyzer insisted I just use `late`.

**Comment by christopherfujino** [comment]
friendly ping @Jasguerrero

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Pull request merged**